### PR TITLE
Get the count of the all system users at endpoint /users/stats

### PR DESCRIPTION
**Pull request by lisakycho**
#### Summary
New api endpoint to get the total number of users in the system. 

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/6521

#### Checklist
- [x] Added API documentation https://github.com/mattermost/mattermost-api-reference/pull/375

### CODE REVIEW

**Comment by jwilander** [comment]
@lisakycho looks like you have a gofmt failure:

```bash
--- /home/ubuntu/jenkins/workspace/mp/mattermost-platform-pr-split/src/github.com/mattermost/mattermost-server/app/user.go.orig	2018-05-23 21:36:19.942182405 +0000
+++ /home/ubuntu/jenkins/workspace/mp/mattermost-platform-pr-split/src/github.com/mattermost/mattermost-server/app/user.go	2018-05-23 21:36:19.942182405 +0000
@@ -1405,7 +1405,6 @@
 	return stats, nil
 }
 
-
 func (a *App) VerifyUserEmail(userId string) *model.AppError {
 	return (<-a.Srv.Store.User().VerifyEmail(userId)).Err
 }
gofmt failure
```

File: api4/user_test.go [hunk 1]
```diff
@@ -909,6 +909,19 @@ func TestGetUsersByUsernames(t *testing.T) {
...
+// 	defer th.TearDown()
+// 	Client := th.Client
+
+// 	rstats, resp := Client.GetTotalUsersStats("")
+// 	CheckNoError(t, resp)
+
+// 	if rstats.TotalUsersCount != 3 {
+// 		t.Fatal("wrong count")
+// 	}
+// }
```
Comment (jwilander) [inline api4/user_test.go]: Test should be added back

File: app/user.go [hunk 2]
```diff
@@ -1392,6 +1392,18 @@ func (a *App) GetVerifyEmailToken(token string) (*model.Token, *model.AppError)
 	}
 }
 
+func (a *App) GetTotalUsersStats() (*model.UsersStats, *model.AppError) {
+	tchan := a.Srv.Store.User().GetTotalUsersCount()
```
Comment (jwilander) [inline app/user.go]: Minor and changing it isn't necessary, but you're not really saving any time by creating the store channel here. Would be fine to just do:

```Go
if result := <-a.Srv.Store.User().GetTotalUsersCount(); ....
```

File: model/users_stats.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+package model
+
+import (
+	"encoding/json"
+	"io"
+)
+
+type UsersStats struct {
+	TotalUsersCount int64 `json:"total_user_count"`
```
Comment (jwilander) [inline model/users_stats.go]: The field name is `TotalUsersCount` but the JSON field is `total_user_count` ("user" vs "users"). Mind making the pluralization match?

**Comment by lisakycho** [comment]
@saturninoabril, @jwilander is there anything else I can do for this PR?

**Pull request merged**